src/Stratis.Bitcoin/Signals/Signals.cs

```diff
@@ -42,36 +55,49 @@ public class Signals : ISignals
         /// <summary>
         /// Initializes the object with newly created instances of signalers.
         /// </summary>
-        public Signals() : this(new Signaler<Block>(), new Signaler<Transaction>())
+        public Signals() : this(new Signaler<Block>(), new Signaler<Block>(), new Signaler<Transaction>())
         {
         }
 
         /// <summary>
         /// Initializes the object with specific signalers.
         /// </summary>
-        /// <param name="blockSignaler">Signaler providing notifications about newly available blocks to its subscribers.</param>
+        /// <param name="blockConnectedSignaler">Signaler providing notifications about newly available blocks to its subscribers.</param>
+        /// <param name="blockDisonnectedSignaler">Signaler providing notifications about a block being disconnected to its subscribers.</param>
         /// <param name="transactionSignaler">Signaler providing notifications about newly available transactions to its subscribers.</param>
-        public Signals(ISignaler<Block> blockSignaler, ISignaler<Transaction> transactionSignaler)
+        public Signals(ISignaler<Block> blockConnectedSignaler, ISignaler<Block> blockDisonnectedSignaler, ISignaler<Transaction> transactionSignaler)
         {
-            Guard.NotNull(blockSignaler, nameof(blockSignaler));
+            Guard.NotNull(blockConnectedSignaler, nameof(blockConnectedSignaler));
```

Comment:
I don't think we would have use case for that. Can you think of any? If you would allow this to be not provided, you would like to specify that the parameter comment that it can be null, but I can't see any reason for that

Reply:
Ok, makes sense. Added a guard for disconnected signal

Reply:
Yes you will always initialise a signaller, it might not have any subscribers but you dont know that at this point